pkg/util/kubernetes/apiserver/leaderelection/leaderelection.go

```diff
@@ -87,17 +90,31 @@ func ResetGlobalLeaderEngine() {
 	telemetryComponent.GetCompatComponent().Reset()
 }
 
-// GetLeaderEngine returns a leader engine client with default parameters.
+// Initialize initializes the leader engine
+func (le *LeaderEngine) Initialize() *retry.Error {
```

Comment:
Is `Initialize` required? I see it only used in `cmd/cluster-agent/subcommands/start/command.go` which could just do a `GetLeaderEngine()`, it would simplify a bit the code and hide the initialization (which we may not want to expose)

Reply:
I added it for the tests and also to avoid doing this in the start command:
```
CreateGlobalLeaderEngine(mainCtx)
le, err := GetLeaderEngine()
```

Reply:
Done as well ! 👍